src/environment.cpp

```diff
@@ -112,10 +112,52 @@ inline static bool isPointableNode(const MapNode &n,
 	       (liquids_pointable && features.isLiquid());
 }
 
+inline static void updateMaxOverlap(f32 overlap, f32 extent, f32& max_overlap, f32& max_overlap_extent)
+{
+	if (overlap > 0 && (max_overlap_extent == 0 || overlap * max_overlap_extent > extent * max_overlap)) {
+		max_overlap = overlap;
+		max_overlap_extent = extent;
+	}
+}
+
 void Environment::continueRaycast(RaycastState *state, PointedThing *result)
 {
 	const NodeDefManager *nodedef = getMap().getNodeDefManager();
 	if (state->m_initialization_needed) {
+		// Check boundaries
+		const float limit = MAX_MAP_GENERATION_LIMIT * BS;
+		if (!std::isfinite(state->m_shootline.start.X) ||
+			!std::isfinite(state->m_shootline.start.Y) ||
+			!std::isfinite(state->m_shootline.start.Z) ||
+			!std::isfinite(state->m_shootline.end.X) ||
+			!std::isfinite(state->m_shootline.end.Y) ||
+			!std::isfinite(state->m_shootline.end.Z) ||
```

Comment:
this is already assured

Reply:
Where?

Reply:
> Where?

Isn't this covered by `std::abs(state->m_shootline.start.X) > limit` below? Infinity is always greater than `limit`.

Reply:
What about NaN?

Reply:
What about `std::isnan` to differentiate those cases easily in code?

Reply:
std::isnan makes code less clear IMO

Reply:
Wrong place to catch this IMO. We expect the engine to only pass finite numbers. And the Lua side of things should get an error thrown if it passes infinities or NaN.